### PR TITLE
Improve dark mode for some preference dialog items

### PR DESCRIPTION
**Pull request by ozone10**
Fix edit control in Date Time for Custom format
Fix tooltip in Delimiter for Word character list
Fix blurry text in Link for Clickable Link Settings

Fix https://github.com/notepad-plus-plus/notepad-plus-plus/issues/11497

### CODE REVIEW

File: PowerEditor/src/WinControls/Preference/preferenceDlg.cpp [hunk 1]
```diff
@@ -3988,6 +3988,15 @@ intptr_t CALLBACK MultiInstanceSubDlg::run_dlgProc(UINT message, WPARAM wParam,
...
 
+		case WM_CTLCOLOREDIT:
+		{
+			if (NppDarkMode::isEnabled())
+			{
+				return NppDarkMode::onCtlColorSofter(reinterpret_cast<HDC>(wParam));
+			}
+			break;
+		}
+
```
Comment (ozone10) [inline PowerEditor/src/WinControls/Preference/preferenceDlg.cpp:3999]: For custom format edit control

File: PowerEditor/src/WinControls/Preference/preferenceDlg.cpp [hunk 2]
```diff
@@ -4245,6 +4254,11 @@ intptr_t CALLBACK DelimiterSubDlg::run_dlgProc(UINT message, WPARAM wParam, LPAR
 
 		case WM_CTLCOLOREDIT:
 		{
+			if (_tip)
+			{
+				NppDarkMode::setDarkTooltips(_tip, NppDarkMode::ToolTipsType::tooltip);
+			}
+
```
Comment (ozone10) [inline PowerEditor/src/WinControls/Preference/preferenceDlg.cpp:4261]: For tooltip

File: PowerEditor/src/WinControls/Preference/preferenceDlg.cpp [hunk 3]
```diff
@@ -4491,6 +4503,36 @@ intptr_t CALLBACK CloudAndLinkSubDlg::run_dlgProc(UINT message, WPARAM wParam, L
...
+					SetTextColor(hdcStatic, textColor);
+				}
+				else
+				{
+					SetTextColor(hdcStatic, disabledTextColor);
+				}
+			}
+			return result;
+		}
+
```
Comment (ozone10) [inline PowerEditor/src/WinControls/Preference/preferenceDlg.cpp:4535]: For blurry text

**Comment by shriprem** [comment]
Nice that you are using the standard `COLOR_GRAYTEXT` and `COLOR_WINDOWTEXT` for light mode instead of the hardcoded RGB() values that were employed to handle blurry text issues elsewhere in NPP.

Personally, I feel the blurry text issue is being handled with repeat code in too many places in NPP. Ideally, this handling deserves its own function in NPPDarMode.cpp. With a signature something like:
```C++
LRESULT onCtlColorIfEnabled(HDC hdc, bool isEnabled);
```
I have an implementation for it at [NPP_Plugin_Drakmode](https://github.com/shriprem/NPP_Plugin_Darkmode/blob/cd37703c8e663c846b7980e1ddda6433d913db3c/NPP_Plugin_Darkmode/NppDarkMode.cpp#L1552). And a usage sample for the [GotoLineCol](https://github.com/shriprem/Goto-Line-Col-NPP-Plugin/blob/f4d52df191f1199f4ec4f0a630be46444b4ff830/src/Dialogs/PreferencesDialog.cpp#L119) plugin.

**Comment by ozone10** [comment]
@shriprem You're right.
I've already had implementation for it but I was waiting for this PR to be merged. 
But it's small change so I've included it in this PR.

Other blurry static texts have issue, they are using color for disabled text from dark mode.
This need to be fixed, but it should be done in another PR.

**Pull request closed without merging**